x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/action/TransportExplainLifecycleAction.java

```diff
@@ -165,9 +185,10 @@ static IndexLifecycleExplainResponse getIndexLifecycleExplainResponse(
 
         final IndexLifecycleExplainResponse indexResponse;
         if (metadata.isIndexManagedByILM(indexMetadata)) {
+            final IndexLifecycleMetadata indexLifecycleMetadata = metadata.custom(IndexLifecycleMetadata.TYPE);
```

Comment:
The index may have ILM metadata, but we could potentially not have any in the cluster state, so we should protect against a NPE with:

```suggestion
            final IndexLifecycleMetadata indexLifecycleMetadata = metadata.custom(IndexLifecycleMetadata.TYPE, IndexLifecycleMetadata.EMPTY);
```